### DISCUSSION TITLE
Contributer

### DISCUSSION THREAD

**Discussion by Dvinod28** [idurar/idurar-erp-crm#221] [General]
How to start contributing to this project, where I can find locally setup guide ?
Can any one help please ?

**Comment by PierreNTE** [comment]
Hi, To start this project, follow these instructions (if you have any problems, you can send me a message):

1 : Git clone the project

2 : Open it with a code editor (I use vscode). You have a tmp.variable.env file. In this file, change "DATABASE = “your-mongodb-uri” to your MongoDB Atlas cluster URI. After that, rename this file to .variable.env

3 : Run “npm i” and “npm start” at idurar-erp-crm. And run “npm i” and “npm start” at idurar-erp-crm/frontend.

This works for me.

**Reply by PierreNTE** [reply]
## Setup Guide

To run this project locally, follow these steps:

**1. Clone the repository:**

```bash
git clone https://github.com/idurar/idurar-erp-crm.git
cd idurar-erp-crm 
```
**Set up your database:**

This project uses MongoDB as the database. You need to create a cluster in MongoDB Atlas and a user with a username and password. Then, edit the `tmp.variables.env` file and replace the `DATABASE` value with your MongoDB URI. For example:

```bash
DATABASE = "mongodb+srv://username:password@clustername.mongodb.net/idurar"
```

After that, rename the file to `.variables.env`

**Install the dependencies and start the project:**

In the `idurar-erp-crm` folder, run:

```bash
npm install
npm start
```

In the `idurar-erp-crm/frontend` folder, run:

```bash
npm install
npm start
```

The project should be running on `http://localhost:3000`

**Reply by FaizAmin99** [reply]
Hi Pierre, thankfully this conversation is here. I have done the same steps; also tried updating the dependencies  to the latest version, as well as clearing the cache and deleting the package.json lock file to create a new one. But nothing worked out.

The backend is working perfectly but on the frontend it is giving the following error:

> erp-crm@0.1.0 start
> craco start

node:internal/modules/cjs/loader:444
      throw err;
      ^

Error: Cannot find module 'D:\React Applications\DymaxTech CRM Sales\frontend\node_modules\isexe\index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:436:19)
    at Module._findPath (node:internal/modules/cjs/loader:678:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1061:27)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (D:\React Applications\DymaxTech CRM Sales\frontend\node_modules\cross-spawn\node_modules\which\which.js:7:15)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32) {
  code: 'MODULE_NOT_FOUND',
  path: 'D:\\React Applications\\DymaxTech CRM Sales\\frontend\\node_modules\\isexe\\package.json',
  requestPath: 'isexe'
}

Node.js v18.16.0

I have also tried using react scripts instead of CRACO, Still no luck :(.

**Reply by polymahh** [reply]
@FaizAmin99  try following this readme guide https://github.com/idurar/idurar-erp-crm/tree/feature/readme-getting-started

**Reply by PierreNTE** [reply]
You can try `npm build` in frontend folden

**Reply by FaizAmin99** [reply]
The 'build' command is giving this error:

> erp-crm@0.1.0 build
> craco build

node:internal/modules/cjs/loader:444
      throw err;
      ^

Error: Cannot find module 'D:\React Applications\DymaxTech CRM Sales\frontend\node_modules\isexe\index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:436:19)
    at Module._findPath (node:internal/modules/cjs/loader:678:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1061:27)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (D:\React Applications\DymaxTech CRM Sales\frontend\node_modules\cross-spawn\node_modules\which\which.js:7:15)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32) {
  code: 'MODULE_NOT_FOUND',
  path: 'D:\\React Applications\\DymaxTech CRM Sales\\frontend\\node_modules\\isexe\\package.json',
  requestPath: 'isexe'
}

Node.js v18.16.0

**Reply by FaizAmin99** [reply]
![image](https://github.com/idurar/idurar-erp-crm/assets/134964161/2b460a09-a3d8-4eab-9668-0a455b1b0e5e)

After react scripts, its still the same issue

**Reply by PierreNTE** [reply]
I’ll get back to you later, I’m doing some tests, but indeed there are some issues, I downloaded the project again and I have the same problems

**Reply by PierreNTE** [reply]
Hi @FaizAmin99,
I redownloaded the repo by following the new instructions (here: https://github.com/idurar/idurar-erp-crm).
Try to redownload it by following these instructions.
(Sorry for the delay in my response)

**Comment by Dvinod28** [comment]
thanks, can we contribute to this project ?

**Comment by salahlalami** [comment]
@Dvinod1  Thank you for your interest to contribute to IDURAR project ,
Here a Contribution Guide : [Contribution](https://github.com/idurar/idurar-erp-crm/blob/master/CONTRIBUTING.md)

**Comment by abhms** [comment]
![image](https://github.com/idurar/idurar-erp-crm/assets/65820065/884fbcb1-d2e7-41b1-8d11-7c60993f8b04)
Hey @salahlalami ,i got this error how can i solve it

**Reply by salahlalami** [reply]
@Dvinod1 please create new issue , and provide more information

**Comment by abhms** [comment]
https://github.com/idurar/erp-crm/pull/582/files#diff-4a2d9aa3e849b134993936ca81b83fb139edd2b0218077ab0f403b8c4803c62a
@salahlalami , this is changed file, i need to revert it or it is able to merge in code ??
